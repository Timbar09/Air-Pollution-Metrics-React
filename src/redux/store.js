import { configureStore } from '@reduxjs/toolkit';

import statsReducers from './stats/statsSlice';

const store = configureStore({
  reducer: {
    stats: statsReducers,
  },
});

export default store;
