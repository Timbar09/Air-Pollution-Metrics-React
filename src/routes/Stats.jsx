import { useSelector } from 'react-redux';

import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { FaChevronLeft } from 'react-icons/fa';

import Header from '../components/Header';
import StatsList from '../components/StatsList';

import styles from '../styles/Stats.module.scss';

const Stats = () => {
  const { country, capital } = useSelector((state) => state.stats);
  const flag = country.toLowerCase().replace(/ /g, '-');

  return (
    <>
      <Header icon={<FaChevronLeft />} />

      <Container className={styles.hero_container}>
        <Row className={styles.hero}>
          <Col>
            <img className={styles.hero_img} src={`flags/${flag}.png`} alt="Flag" />
          </Col>
          <Col className={styles.hero_title}>
            <div>
              <h1>{country}</h1>
              <span>{capital}</span>
            </div>
          </Col>
        </Row>
      </Container>

      <StatsList />
    </>
  );
};

export default Stats;
