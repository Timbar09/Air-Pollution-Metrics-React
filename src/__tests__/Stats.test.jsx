import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Stats from '../routes/Stats';
import store from '../redux/store';

describe('Stats tests', () => {
  test('should render Stats page', () => {
    const stats = render(
      <Provider store={store}>
        <BrowserRouter>
          <Stats />
        </BrowserRouter>
      </Provider>,
    );

    expect(stats).toMatchSnapshot();
  });
});
