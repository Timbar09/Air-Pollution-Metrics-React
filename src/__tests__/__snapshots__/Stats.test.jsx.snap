// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stats tests > should render Stats page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_a5b446 py-2"
      >
        <div
          class="container"
        >
          <nav
            class="_header_nav_a5b446"
          >
            <a
              aria-current="page"
              class="_header_link_a5b446 active"
              href="/"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
                />
              </svg>
            </a>
            <span>
              Air Pollution Metrics
            </span>
            <button
              class="_header_link_a5b446"
              title="Select search filter"
              type="button"
            >
              <svg
                class="_header_gear_a5b446"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
                />
              </svg>
            </button>
          </nav>
        </div>
      </header>
      <div
        class="_hero_container_ffeb6e container"
      >
        <div
          class="_hero_ffeb6e row"
        >
          <div
            class="col"
          >
            <img
              alt="Flag"
              class="_hero_img_ffeb6e"
              src="flags/.png"
            />
          </div>
          <div
            class="_hero_title_ffeb6e col"
          >
            <div>
              <h1 />
              <span />
            </div>
          </div>
        </div>
      </div>
      <div>
        Latitude and longitude unavailable.
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_a5b446 py-2"
    >
      <div
        class="container"
      >
        <nav
          class="_header_nav_a5b446"
        >
          <a
            aria-current="page"
            class="_header_link_a5b446 active"
            href="/"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
              />
            </svg>
          </a>
          <span>
            Air Pollution Metrics
          </span>
          <button
            class="_header_link_a5b446"
            title="Select search filter"
            type="button"
          >
            <svg
              class="_header_gear_a5b446"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.405 1.05c-.413-1.4-2.397-1.4-2.81 0l-.1.34a1.464 1.464 0 0 1-2.105.872l-.31-.17c-1.283-.698-2.686.705-1.987 1.987l.169.311c.446.82.023 1.841-.872 2.105l-.34.1c-1.4.413-1.4 2.397 0 2.81l.34.1a1.464 1.464 0 0 1 .872 2.105l-.17.31c-.698 1.283.705 2.686 1.987 1.987l.311-.169a1.464 1.464 0 0 1 2.105.872l.1.34c.413 1.4 2.397 1.4 2.81 0l.1-.34a1.464 1.464 0 0 1 2.105-.872l.31.17c1.283.698 2.686-.705 1.987-1.987l-.169-.311a1.464 1.464 0 0 1 .872-2.105l.34-.1c1.4-.413 1.4-2.397 0-2.81l-.34-.1a1.464 1.464 0 0 1-.872-2.105l.17-.31c.698-1.283-.705-2.686-1.987-1.987l-.311.169a1.464 1.464 0 0 1-2.105-.872l-.1-.34zM8 10.93a2.929 2.929 0 1 1 0-5.86 2.929 2.929 0 0 1 0 5.858z"
              />
            </svg>
          </button>
        </nav>
      </div>
    </header>
    <div
      class="_hero_container_ffeb6e container"
    >
      <div
        class="_hero_ffeb6e row"
      >
        <div
          class="col"
        >
          <img
            alt="Flag"
            class="_hero_img_ffeb6e"
            src="flags/.png"
          />
        </div>
        <div
          class="_hero_title_ffeb6e col"
        >
          <div>
            <h1 />
            <span />
          </div>
        </div>
      </div>
    </div>
    <div>
      Latitude and longitude unavailable.
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
